libvast/src/system/type_registry.cpp

```diff
@@ -111,27 +130,65 @@ caf::error type_registry_state::load_from_disk() {
     VAST_DEBUG("{} found no directory to load from", *self);
     return caf::none;
   }
-  const auto fname = filename();
-  const auto file_exists = std::filesystem::exists(fname, err);
-  if (err)
-    return caf::make_error(ec::filesystem_error,
-                           fmt::format("failed to find file {}: {}", fname,
-                                       err.message()));
-  if (file_exists) {
-    auto buffer = io::read(fname);
-    if (!buffer)
-      return buffer.error();
-    std::map<std::string, detail::stable_set<legacy_type>> intermediate = {};
-    if (!detail::legacy_deserialize(*buffer, intermediate))
-      return caf::make_error(ec::parse_error, "failed to load type-registry "
-                                              "state");
-    for (const auto& [k, vs] : intermediate) {
-      auto entry = type_set{};
-      for (const auto& v : vs)
-        entry.emplace(type::from_legacy_type(v));
-      data.emplace(k, entry);
+  // Support the legacy CAF-serialized state, and delete it afterwards.
+  {
+    const auto fname = dir / name;
+    const auto file_exists = std::filesystem::exists(fname, err);
+    if (err)
+      return caf::make_error(ec::filesystem_error,
+                             fmt::format("failed to find file {}: {}", fname,
```

Comment:
We can make this message a little clearer, the sentence can me misinterpreted to mean that the file was not found.
```suggestion
                             fmt::format("failed while trying to find file {}: {}", fname,
```